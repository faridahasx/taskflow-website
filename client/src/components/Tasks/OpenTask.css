.open-task-wrapper {
    min-height: calc(var(--vh) - 80px);
    max-height: calc(var(--vh) - 80px);
    overflow: auto;
    padding: 0 5px;
}

.open-task-wrapper .task-dialog-title {
    max-height: 100%;
    padding: 10px 20px;
    margin-top: 5px;
    margin-bottom: 20px;
    background-color: rgba(234, 234, 234, 0.442);
    overflow-wrap: break-word;
    font-size: 19px;
    word-wrap: break-word;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
}


.task-description {
    width: 100%;
    overflow-wrap: break-word;
}

.t-icon {
    min-width: 40px;
    min-height: 40px;
    color: rgb(59, 59, 59);
}

.description-loading {
    min-height: calc(var(--vh) - 200px);
}

@media all and (min-width: 900px) and (min-height: 500px) {
    .open-task-wrapper {
        min-height: calc(((var(--vh) / 100) * 88) - 90px);
        max-height: calc(((var(--vh) / 100) * 88) - 90px);
    }

    .description-loading {
        min-height: calc(((var(--vh) / 100) * 88) - 200px);
    }
}