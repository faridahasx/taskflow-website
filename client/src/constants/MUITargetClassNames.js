export default [
  "PrivateSwitchBase-input",
  "MuiInputBase-input",
  "MuiSvgIcon-root",
  "MuiPickersArrowSwitcher-root",
  "MuiButtonBase-root",
  "MuiPickersSlideTransition-root",
  "MuiModal-backdrop",
  "MuiTypography-root",
  "MuiButtonBase-root",
  "MuiPaper-root",
  "MuiSvgIcon-root",
  "MuiPickersCalendarHeader-label",
  "MuiDateCalendar-root",
  "MuiDateRangePickerDay-rangeIntervalPreview",
  "MuiList-root",
  "MuiDayCalendar-weekContainer",
  "MuiDateRangeCalendar-monthContainer",
  "MuiDayCalendar-header",
  "MuiButtonBase-root",
  "MuiDialogActions-root",
  "MuiPickersCalendarHeader-root",
  "MuiButtonBase-root",
  "MuiDayCalendar-monthContainer",
  "MuiPickersYear-yearButton",
  "MuiPickersYear-root",
  "MuiYearCalendar-root",
  "MuiPickersCalendarHeader-labelContainer",
];
