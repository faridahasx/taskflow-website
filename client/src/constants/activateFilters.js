const activateFilters = [
  { buttonTxt: "All", value: "all" },
  { buttonTxt: "Current", value: "current" },
  { buttonTxt: "Overdue", value: "overdue" },
  { buttonTxt: "Upcoming", value: "upcoming" },
  { buttonTxt: "Completed", value: "completed" },
];

export default activateFilters;
