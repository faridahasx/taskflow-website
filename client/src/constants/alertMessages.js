// Categories
export const CATEGORY_DELETED = "Category Deleted";
export const DELETE_CATEGORY_FAILED = "Failed to delete the category";

// Tasks
export const TASK_DELETED = "Task Deleted";
export const DELETE_TASK_FAILED = "Failed to delete the task";
export const FETCH_TASKS_FAILED = "Failed to fetch tasks";

// Generic
export const PERMANENT_ACTION = "This action cannot be undone";
export const MISSING_INPUT_FIELD = "Please fill out the required fields";

// Errors
export const UNKNOWN_ERROR = "Something went wrong";
export const NO_INTERNET = "Please check your internet connection";
export const UNAUTHORIZED = "Please login";
