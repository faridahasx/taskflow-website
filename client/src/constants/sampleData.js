import { getDateWithNDaysOfDifference } from "../utils/dateUtils";

const categoriesSample = [
  {
    title: "All",
    tasks: 5,
    _id: "All",
  },
  {
    _id: "Tasks",
    tasks: 5,
    title: "Tasks",
  },
];

let tasksSample = [
  {
    _id: "1",
    title: "Task 1",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(-1),
    finishDate: getDateWithNDaysOfDifference(1),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
  {
    _id: "2",
    title: "Task 2",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(-2),
    finishDate: getDateWithNDaysOfDifference(-1),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
  {
    _id: "3",
    title: "Task 3",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(2),
    finishDate: getDateWithNDaysOfDifference(3),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },

  {
    _id: "4",
    title: "Task 4",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(2),
    finishDate: getDateWithNDaysOfDifference(4),
    completedAt: getDateWithNDaysOfDifference(3),
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
  {
    _id: "5",
    title: "Login to continue",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(-1),
    finishDate: getDateWithNDaysOfDifference(1),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
];

export { tasksSample, categoriesSample };
