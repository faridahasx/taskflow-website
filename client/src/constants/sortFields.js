const sortFields = [
  { name: "First start first", value: "startDate:1" },
  { name: "Last start first", value: "startDate:-1" },
  { name: "First finish first", value: "finishDate:1" },
  { name: "Last finish first", value: "finishDate:-1" },
  { name: "First update first", value: "updatedAt:1" },
  { name: "Last update first", value: "updatedAt:-1" },
  { name: "First created first", value: "createdAt:1" },
  { name: "Last created first", value: "createdAt:-1" },
];

export default sortFields;
