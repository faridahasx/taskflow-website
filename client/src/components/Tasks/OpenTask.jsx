// External imports
import { useEffect } from "react";
import PropTypes from "prop-types";
// Components
import CircularLoading from "../Loading/CircularLoading";
import EditButton from "../IconButtons/EditButton";
import TaskDialog from "./TaskDialog";
// Styles
import "./OpenTask.css";
import "./TaskForm/TextEditor/styles.css";

const OpenTask = (props) => {
  // Destructure props
  const {
    task,
    openEditTaskEditor,
    handleClickEditTask,
    handleCloseOpenedTask,
    loading,
  } = props;

  // useEffect to manage body class for overflow
  useEffect(() => {
    document.body.classList.add("overflow-hidden");
    return () => {
      !openEditTaskEditor._id &&
        document.body.classList.remove("overflow-hidden");
    };
  }, [openEditTaskEditor]);

  return (
    <TaskDialog
      handleBackgroundClick={handleCloseOpenedTask}
      handleCloseClick={handleCloseOpenedTask}
      SecondButton={<EditButton onClick={handleClickEditTask} />}
    >
      <div className="open-task-wrapper">
        <h1 className="task-dialog-title flex">{task.title}</h1>
        <div className="center">
          {loading ? (
            <div className="description-loading center">
              <CircularLoading />
            </div>
          ) : (
            <section
              className="task-description"
              dangerouslySetInnerHTML={{ __html: task.description }}
            />
          )}
        </div>
      </div>
    </TaskDialog>
  );
};

OpenTask.propTypes = {
  task: PropTypes.object.isRequired,
  openEditTaskEditor: PropTypes.object,
  handleClickEditTask: PropTypes.func.isRequired,
  handleCloseOpenedTask: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default OpenTask;
