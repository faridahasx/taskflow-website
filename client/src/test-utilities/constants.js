import { MISSING_INPUT_FIELD } from "constants/alertMessages";

// AUTH
export const passwordLabel = "Password:";
export const emailLabel = "Email:";
export const firstnameLabel = "First Name:";
export const lastnameLabel = "Last Name:";

// Server responses
export const successResponse = { status: 200, data: "Success" };

// Redux state
export const reduxStateOnMissingInputField = [
  {
    type: "ALERT",
    payload: MISSING_INPUT_FIELD,
  },
];
