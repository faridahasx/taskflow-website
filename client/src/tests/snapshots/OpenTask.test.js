import { render } from "@testing-library/react";
import OpenTask from "../../components/Tasks/OpenTask";
import { tasksSample } from "../../constants/sampleData";

const mockFn = jest.fn();

test("OpenTask snapshot", () => {
  const { asFragment } = render(
    <OpenTask
      task={tasksSample[0]}
      openEditTaskEditor={{}}
      handleClickEditTask={mockFn}
      handleCloseOpenedTask={mockFn}
      loading={false}
    />
  );

  expect(asFragment()).toMatchSnapshot("OpenTask");
});
