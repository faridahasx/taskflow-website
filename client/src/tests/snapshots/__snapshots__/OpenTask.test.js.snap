// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenTask snapshot: OpenTask 1`] = `
<DocumentFragment>
  <div
    class="modal task-modal"
  >
    <div
      class="task-dialog"
    >
      <div
        class="flex task-dialog-top"
      >
        <button
          class="icon "
          title="Close"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
        <button
          class="icon "
          title="Edit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EditIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </button>
      </div>
      <div
        class="open-task-wrapper"
      >
        <h1
          class="task-dialog-title flex"
        >
          Task 1
        </h1>
        <div
          class="center"
        >
          <section
            class="task-description"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
